src/components/Frame/components/Loading/Loading.tsx

```diff
@@ -1,79 +1,50 @@
-import React, {Component} from 'react';
-import debounce from 'lodash/debounce';
+import React, {useEffect, useState} from 'react';
 
-import styles from './Loading.scss';
-
-export interface LoadingProps {}
+import {useI18n} from '../../../../utilities/i18n';
 
-interface State {
-  progress: number;
-  step: number;
-  animation: number | null;
-}
+import styles from './Loading.scss';
 
 const INITIAL_STEP = 10;
 const STUCK_THRESHOLD = 99;
 
-export class Loading extends Component<LoadingProps, State> {
-  state: State = {
-    progress: 0,
-    step: INITIAL_STEP,
-    animation: null,
-  };
-
-  private ariaValuenow = debounce(() => {
-    const {progress} = this.state;
-    return Math.floor(progress / 10) * 10;
-  }, 15);
-
-  componentDidMount() {
-    this.increment();
-  }
-
-  componentWillUnmount() {
-    const {animation} = this.state;
-
-    if (animation != null) {
-      cancelAnimationFrame(animation);
-    }
-  }
-
-  render() {
-    const {progress} = this.state;
-
-    const customStyles = {
-      transform: `scaleX(${Math.floor(progress) / 100})`,
-    };
-
-    const ariaValuenow = this.ariaValuenow();
-
-    return (
-      <div className={styles.Loading}>
-        <div
-          className={styles.Level}
-          style={customStyles}
-          aria-valuenow={ariaValuenow}
-          aria-valuemin={0}
-          aria-valuemax={100}
-          role="progressbar"
-        />
-      </div>
-    );
-  }
+export function Loading() {
+  const i18n = useI18n();
 
-  private increment() {
-    const {progress, step} = this.state;
+  const [progress, setProgress] = useState(0);
+  const [step, setStep] = useState(INITIAL_STEP);
+  const [animating, setAnimating] = useState(false);
 
-    if (progress >= STUCK_THRESHOLD) {
+  useEffect(() => {
+    if (progress >= STUCK_THRESHOLD || animating) {
       return;
     }
 
-    const animation = requestAnimationFrame(() => this.increment());
-
-    this.setState({
-      progress: Math.min(progress + step, 100),
-      step: INITIAL_STEP ** -(progress / 25),
-      animation,
+    requestAnimationFrame(() => {
```

Comment:
This `rAF` is needed to trigger the transition otherwise it sets the value to `10` and does not animate between 0 and 10.